mono/metadata/appdomain.c

```diff
@@ -2431,6 +2431,7 @@ unload_thread_main (void *arg)
 	/* Force it to be attached to avoid racing during shutdown. */
 	thread = mono_thread_attach_full (mono_get_root_domain (), TRUE, &error);
 	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	mono_thread_set_name_internal (thread->internal_thread, mono_string_new (mono_domain_get (), "Domain unloader"), TRUE);
```

Comment:
This is wrong, the thead is attached to the root domain so the name must be allocated there too.